build-scripts/build_course_zips.js

```diff
@@ -45,108 +43,115 @@ const distPath = options.dist
 const coursesPath = options.courses
 const zipsPath = options.zips
 
-// clear out the distribution path and run the webpack build
-rimraf.sync(distPath)
-if (shell.exec("npm run build:webpack").code !== 0) {
-  console.error("Webpack build failed")
-  process.exit(1)
+
+// Walk a tree and produce { root, relPath, file } for each file found
+async function* iterateTree(rootPath, relPath = ".") {
+  const files = await fsPromises.readdir(path.join(rootPath, relPath))
+  for (const file of files) {
+    const absFile = path.join(rootPath, relPath, file)
+    const stat = await fsPromises.lstat(absFile)
+    if (stat.isDirectory()) {
+      for await (const item of iterateTree(rootPath, path.join(relPath, file))) {
+        yield item
+      }
+    } else if (stat.isFile()) {
+      yield { root: rootPath, relPath, file }
+    }
+  }
 }
 
-// remove existing zips
-rimraf.sync(zipsPath)
-fs.mkdirSync(zipsPath, { recursive: true })
+// clear out the distribution path and run the webpack build
+const run = async () => {
+  if ((await execFile("which", ["zip"])).error) {
+    throw new Error("Unable to find zip binary")
+  }
+
+  const baseDir = path.resolve(coursesPath, "..", "..")
+  const relative = path.relative(baseDir, zipsPath)
+  if (relative && !relative.startsWith("..") && !path.isAbsolute(relative)) {
+    // make sure zips are not picked up by hugo, causing a blowup in file size
+    throw new Error("zips path must not be within hugo course area, two parent directories above courses directory")
```

Comment:
This check should already handle that case since it checks for `zips` anywhere within two parent directories of the course directory, which is anything within `site` by default. Would that cover everything?

Reply:
Ah yes, this is correct and that should cover both cases.